### ISSUE TITLE
[CI] FollowIndexSecurityIT testUnPromoteAndFollowDataStream failing

### ISSUE THREAD

**Issue by mark-vieira** [elastic/elasticsearch#70942]
Looks like this test failed (along with a couple of others in the same build) when executed right around midnight UTC. Clock issue?

**Build scan**:
https://gradle-enterprise.elastic.co/s/uicegqvfk46bq/tests/:x-pack:plugin:ccr:qa:security:follow-cluster/org.elasticsearch.xpack.ccr.FollowIndexSecurityIT/testUnPromoteAndFollowDataStream#1

**Repro line**:
`./gradlew ':x-pack:plugin:ccr:qa:security:follow-cluster' --tests "org.elasticsearch.xpack.ccr.FollowIndexSecurityIT.testUnPromoteAndFollowDataStream" -Dtests.seed=5EFEF2509AAD91AB -Dtests.security.manager=true -Dtests.locale=es-SV -Dtests.timezone=Asia/Aqtau -Druntime.java=8`

**Reproduces locally?**:
No.

**Applicable branches**:
`7.x` (possibly master as well?)

**Failure history**:
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.ccr.FollowIndexSecurityIT&tests.test=testUnPromoteAndFollowDataStream

**Failure excerpt**:
```
org.elasticsearch.xpack.ccr.FollowIndexSecurityIT > testUnPromoteAndFollowDataStream FAILED
    java.lang.AssertionError: 
    Expected: ".ds-logs-eu-monitor1-2021.03.27-000001"
         but: was ".ds-logs-eu-monitor1-2021.03.26-000001"
        at __randomizedtesting.SeedInfo.seed([5EFEF2509AAD91AB:52B0F87C8A1AFFEE]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.ccr.ESCCRRestTestCase.verifyDataStream(ESCCRRestTestCase.java:319)
        at org.elasticsearch.xpack.ccr.FollowIndexSecurityIT.testUnPromoteAndFollowDataStream(FollowIndexSecurityIT.java:284)
```

**Comment by jtibshirani** [comment]
This may be another instance: https://gradle-enterprise.elastic.co/s/azne45knxsl5a. It happened right around midnight UTC.

Failure excerpt:
```
   {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [.ds-logs-eu-monitor1-2021.04.22-000001]","index_uuid":"_na_","index":".ds-logs-eu-monitor1-2021.04.22-000001"}],"type":"index_not_found_exception","reason":"no such index [.ds-logs-eu-monitor1-2021.04.22-000001]","index_uuid":"_na_","index":".ds-logs-eu-monitor1-2021.04.22-000001"},"status":404}
        at __randomizedtesting.SeedInfo.seed([69371510C7F7E385:65791F3CD7408DC0]:0)
        at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:326)
```

**Comment by martijnvg** [comment]
This failure was also fixed via #71419 a while ago and as far as I can tell hasn't failed since.
